Zend/zend_vm_def.h

```diff
@@ -4271,7 +4299,9 @@ ZEND_VM_C_LABEL(fcall_end):
 		HANDLE_EXCEPTION();
 	}
 
-	ZEND_VM_SET_OPCODE(opline + 1);
+	// todo: under what circumstances do ZEND_USER_FUNCTIONs take this code path?
```

Comment:
For `ZEND_DO_FCALL`, this seems to get taken when `zend_execute_ex` is set to something other than `execute_ex`. Otherwise, it seems the same as my other todo (`ZEND_VM_ENTER_EX` gets expanded into a return or jmp, so this code is never reached).

If that assumption about `ZEND_VM_ENTER_EX` is correct, then I suppose the fix is to do the interrupt at the end of the else branch of `if (EXPECTED(zend_execute_ex == execute_ex))`.

If the assumption is wrong, then I need guidance.

Reply:
I suppose you need to check interrupt only for internal functions.

Reply:
Do you mean that for the case of a user function which takes the `zend_execute_ex` apth, we don't need an interrupt check at all in this opcode handler?